chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java

```diff
@@ -855,14 +855,20 @@ public void addMessage(@NonNull Message message,
      *
      * @param message the message to be edited
      * @param body    the new message body
+     * @param metadata  the new message metadata
+     * @param asset  the new message asset
      * @param callback save callback
      */
 
     public void editMessage(@NonNull Message message,
                             @NonNull String body,
+                            @Nullable final JSONObject metadata,
+                            @Nullable final Asset asset,
                             @Nullable final SaveMessageCallback callback)
     {
         message.setBody(body);
+        message.setMetadata(metadata);
+        message.setAsset(asset);
```

Comment:
Just an idea, do you know it would be better if we keep the original `editMessage` and add another `editMessage` with metadata and asset? Since in this `editMessage`, metadata and asset are required. But in some simple use case. When users only do text only chat, they don't need to always provide meta or asset when editing message. What do you think? @howawong